File: July2021/ReshapeTheMatrix.java
class Solution {

  // TC : O(r*c)
  // SC  : O(r*c)
  public int[][] matrixReshape(int[][] mat, int r, int c) {

    if (mat.length * mat[0].length != r * c)  {
      return mat;
    }
    // the equalit is met
    int[][] newMatrix = new int[r][c];

    int sR = 0;
    int sC = 0;

    for(int i=0;i<mat.length;i++){
      for(int j=0;j<mat[0].length;j++){
        if(sC == c){
          sR++;
          sC = 0;
        }
        newMatrix[sR][sC] = mat[i][j];
        sC++;
      }
    }

    return newMatrix;
  }
}

File: July2021/C++/ReshapeTheMatrix.cpp
class Solution
{
public:
    vector<vector<int>> matrixReshape(vector<vector<int>> &mat, int r, int c)
    {
        int m = size(mat), n = size(mat[0]), total = m * n;
        if (r * c != total)
            return mat;
        vector<vector<int>> ans(r, vector<int>(c));
        for (int i = 0; i < total; i++)
            ans[i / c][i % c] = mat[i / n][i % n];
        return ans;
    }
};